engine/storage/datamotion/src/main/java/org/apache/cloudstack/storage/motion/StorageSystemDataMotionStrategy.java

```diff
@@ -1675,6 +1737,64 @@ private SnapshotDetailsVO handleSnapshotDetails(long csSnapshotId, String value)
         return _snapshotDetailsDao.persist(snapshotDetails);
     }
 
+    /**
```

Comment:
this file is growing over 300 lines. Can we move the KVM live migration code to a TacticalKVMMigrationAid class? (name just a suggestion)